src/main/scala/PopModel.scala

```diff
@@ -18,131 +18,203 @@
-  def calcPopular(appName: String, eventNames: List[String] = List.empty,
-    interval: Interval)(implicit sc: SparkContext): Option[RDD[(String, Float)]] = {
+  def calcPopular(
+    appName: String,
+    eventNames: Seq[String],
+    interval: Interval)(implicit sc: SparkContext): RDD[(ItemID, Float)] = {
 
     val events = eventsRDD(appName, eventNames, interval)
-    val retval = events.map { e => (e.targetEntityId, e.event) }
+    events.map { e => (e.targetEntityId, e.event) }
       .groupByKey()
-      .map { case(itemID, itEvents) => (itemID.get, itEvents.size.toFloat)}
-      .reduceByKey (_+_) // make this a double in Elaseticsearch)
-    if (!retval.isEmpty()) Some(retval) else None
+      .map { case (itemID, itEvents) => (itemID.get, itEvents.size.toFloat) }
+      .reduceByKey(_ + _) // make this a double in Elaseticsearch)
   }
 
   /** Creates a rank for each item by dividing the duration in two and counting named events in both buckets
-    * then dividing most recent by less recent. This ranks by change in popularity or velocity of populatiy change.
-    * Interval(start, end) end instant is always greater than or equal to the start instant.
-    */
-  def calcTrending(appName: String, eventNames: List[String] = List.empty,
-    interval: Interval)(implicit sc: SparkContext): Option[RDD[(String, Float)]] = {
+   *  then dividing most recent by less recent. This ranks by change in popularity or velocity of populatiy change.
+   *  Interval(start, end) end instant is always greater than or equal to the start instant.
+   */
+  def calcTrending(
+    appName: String,
+    eventNames: Seq[String],
+    interval: Interval)(implicit sc: SparkContext): RDD[(ItemID, Float)] = {
 
-    val olderInterval = new Interval(interval.getStart,
-      interval.getStart().plusMillis((interval.toDurationMillis/2)toInt))
-    val newerInterval = new Interval(interval.getStart().plusMillis((interval.toDurationMillis/2)toInt), interval.getEnd)
+    val halfInterval = (interval.toDurationMillis / 2).toInt
+    val olderInterval = new Interval(interval.getStart, interval.getStart.plusMillis(halfInterval))
+    val newerInterval = new Interval(interval.getStart.plusMillis(halfInterval), interval.getEnd)
 
-    val intervalMillis = interval.toDurationMillis
-    val olderPopRDD = calcPopular(appName, eventNames, olderInterval)
-    if ( olderPopRDD.nonEmpty) {
+    val olderPopRDD: RDD[(ItemID, Float)] = calcPopular(appName, eventNames, olderInterval)
+
+    // TODO: need refactor this
+    if (!olderPopRDD.isEmpty()) {
       val newerPopRDD = calcPopular(appName, eventNames, newerInterval)
-      if ( newerPopRDD.nonEmpty ) {
-        val retval = newerPopRDD.get.join(olderPopRDD.get).map { case (item, (newerScore, olderScore)) =>
-          val velocity = (newerScore - olderScore)
-          (item, velocity)
+      if (!newerPopRDD.isEmpty()) {
+        newerPopRDD.join(olderPopRDD).map {
```

Comment:
I believe if we don't check `isEmpty` we will create a `sc.emptyRDD` and `isEmpty` on an RDD is a very heavy weight function. This is my mistake originally.